cpu/atxmega/periph/pm.c

```diff
@@ -78,3 +109,27 @@ void pm_set(unsigned mode)
     sleep_disable();
     irq_restore(irq_state);
 }
+
+void pm_periph_enable(pwr_reduction_t pwr, bool enable)
+{
+    uint8_t mask = _device_mask(pwr);
+    uint8_t *reg = _register_addr(pwr);
+
+    if (enable) {
+        *reg &= ~mask;      /* Clear to Enable */
+    }
+    else {
+        *reg |= mask;
+    }
+}
```

Comment:
I have the feeling that the function call overhead here is greater than the implementation. So if this would be provided as `static inline` instead, I think ROM would be reduced.

If that is the case, would you mind to split this into `pm_periph_enable()` and `pm_periph_disable()`, as this would make reading the code much easier.

If not, maybe you could name the C function to e.g. `pm_periph_set_enabled()` and provide a `static inline pm_periph_enable()` and `static inline pm_periph_disable()` as wrapper? The compiler will optimize the wrappers out, so this would come with no cost.

Reply:
I splitted the method as suggested. I built with/without static and noted that flash savings were higher using the method as not static. Because of that I preferred left two non static methods.